### PR TITLE
compiler: make Stub final class

### PR DESCRIPTION
**Pull request by dapengzhang0**
forgot to make Stub final class...

### CODE REVIEW

File: compiler/src/java_plugin/cpp/java_generator.cpp [hunk 1]
```diff
@@ -541,7 +541,7 @@ static void PrintStub(
     } else {
       p->Print(
           *vars,
-          "public static class $stub_name$ extends $AbstractStub$<$stub_name$> {\n");
+          "public static final class $stub_name$ extends $AbstractStub$<$stub_name$> {\n");
```
Comment (ejona86) [inline compiler/src/java_plugin/cpp/java_generator.cpp]: Should this also be triggered by the deprecated api flag?

Comment (dapengzhang0) [inline compiler/src/java_plugin/cpp/java_generator.cpp]: deprecated stub apis should still be without `final` keyword as some users are using mockito.

Comment (ejona86) [inline compiler/src/java_plugin/cpp/java_generator.cpp]: Oh, I didn't see this was only done if `!enable_deprecated`

**Comment by ejona86** [comment]
@dapengzhang0 LGTM

**Comment by cmargiol** [comment]
Hey guys, 

We had implemented a "RetryableClient" wrapping generated stubs to reconnect dropped connections.  Originally we were using java.lang.reflect.Proxy when there were interfaces, and thought we'd do it the mockito way now for the generated classes. Making them final means we can't - but wouldn't want to enable the deprecated API to use this. 

So I was wondering what is the context for this change?

**Comment by ejona86** [comment]
See https://github.com/grpc/grpc-java/issues/1469#issuecomment-226873511

For your use case, we'd recommend making a `ClientInterceptor`.

**Pull request merged**